draft-ietf-rtcweb-jsep.xml

```diff
@@ -1977,13 +1977,224 @@ candidate:1 1 UDP 1694498815 192.0.2.33 10000 typ host
+            <t>Next, the SessionDescription is parsed into a data structure,
+              as described in the
+              <xref target="sec.parsing-a-desc"/> section below. If parsing
+              fails for any reason, processing MUST stop and an error MUST be
+              returned.
+            </t>
+            <t>Finally, the parsed SessionDescription is applied as described in
+              the <xref target="sec.applying-a-remote-desc"/> section below.</t>
+          </list>
+        </t>
+      </section>
 
-      <section title="Applying a Local Description" anchor="sec.applying-a-local-descript"></section>
+      <section title="Parsing a Session Description" anchor="sec.parsing-a-desc">
+        <t>When a SessionDescription of any type is supplied to
+        setLocal/RemoteDescription, the implementation must parse it and
+        reject it if it is invalid. The exact details of this
+        process are explained below.</t>
+        <t>The SDP contained in the session description object consists of a
+        sequence of text lines, each containing a key-value expression,
+        as described in <xref target="RFC4566"/>, Section 5.
+        The SDP is read,
+        line-by-line, and converted to a data structure that contains the
+        deserialized information. However, SDP allows many types of lines, not
+        all of which are relevant to JSEP applications, and the ones that are
+        not relevant can be ignored.
+        For each line, the implementation will first ensure it is syntactically
+        correct, and then either parse the data value to store for later use,
+        or ignore the value and move on. The implementation MAY store ignored
+        lines for diagnostic purposes or other reasons, but the data contained
+        within will not be used in any future JSEP processing. If the line is
+        not well-formed, or cannot be parsed as described, the parser MUST stop
+        with an error and reject the session decription. This ensures that
+        implementations do not accidentally misinterpret ambiguous SDP.</t>
+
+        <section title="Session-Level Parsing">
+          <t>First, the session-level lines are checked and parsed. These
+            lines MUST occur in a specific order, and with a specific syntax, as
+            defined in <xref target="RFC4566"/>, Section 5. Note that while
+            the specific line types (e.g. "v=", "c=") MUST occur in the defined
+            order, lines of the same type (typically "a=") can occur in any order,
+            and their ordering is not meaningful.</t>
+          <t>For non-attribute (non-"a=") lines, their sequencing and syntax are checked,
+            as mentioned above, but all values MUST be discarded. For example,
+            because the use of ICE is mandatory, the "c=" line is superfluous and its
+            value MUST be discarded; similarly, because session-level bandwidth is
```

Comment:
OK. In the createOffer section, I do explain that this must be very carefully generated. I could add some text to avoid confusion here if you think that would help.